### PR TITLE
Retrieve surahs from API.

### PR DESCRIPTION
**Pull request by govza**
Home Page template almost ready.

Retrieve surahs from API.
Refactor angular models. Play button.

### CODE REVIEW

**Comment by govza** [comment]
@avatsaev Alhamdulillah! Here you are, hope you can understand where I'am & where to go. For quick chat please use [discord](https://discord.gg/A2ZuGrd).

**Comment by avatsaev** [comment]
Great, ill have some free time to look at it tomorrow inshallah

File: frontend/angular/src/app/_services/rest.service.ts [hunk 1]
```diff
@@ -16,4 +17,8 @@ export class RestService {
   getReciters(): Observable<Array<Reciter>> {
     return this.http.get<Array<Reciter>>(endpoint + 'reciters/');
```
Comment (avatsaev) [inline frontend/angular/src/app/_services/rest.service.ts]: use template interpolation https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: frontend/angular/src/app/home/home.component.html [hunk 2]
```diff
@@ -6,5 +6,18 @@
         <mat-option *ngFor="let Reciter of reciters" value="reciter.id">{{ Reciter.name }}</mat-option>
       </mat-select>
     </mat-form-field>
+    <mat-form-field *ngIf = "surahs" appearance="outline">
+      <mat-label>Select Surah</mat-label>
+      <mat-select placeholder="Surah" formControlName="surah">
+        <mat-option *ngFor="let Surah of surahs" value="surah.number">
```
Comment (avatsaev) [inline frontend/angular/src/app/home/home.component.html]: use trackBy for better performance https://netbasal.com/angular-2-improve-performance-with-trackby-cc147b5104e5

File: frontend/angular/src/app/home/home.component.ts [hunk 3]
```diff
@@ -29,4 +32,13 @@ export class HomeComponent implements OnInit {
           console.log('error', error);
         }
   }
+
+  showSurahs() {
+    this.rest.getSurahs()
```
Comment (avatsaev) [inline frontend/angular/src/app/home/home.component.ts:37]: do not subscribe in component controllers, this leads to memory leaks if you don't handle the subscriptions, use async pipe in templates when possible

File: frontend/angular/src/app/_services/rest.service.ts [hunk 4]
```diff
@@ -16,4 +17,8 @@ export class RestService {
...
   }
+
+  getSurahs(): Observable<Array<Surah>> {
```
Comment (avatsaev) [inline frontend/angular/src/app/_services/rest.service.ts:21]: use template literals for interpolation

**Comment by govza** [comment]
@avatsaev let me know if you can't do this. Anyhow, we should go forward.

**Comment by avatsaev** [comment]
Hey, I've already reviewed this

**Comment by govza** [comment]
@avatsaev Ah, thanks, man. But no comments, no approval ? Is it ok so far? :)

**Review by avatsaev — COMMENTED** [review]
> @avatsaev Ah, thanks, man. But no comments, no approval ? Is it ok so far? :)

I've added several comments highlighting some issues already, can't you see them?

**Comment by avatsaev** [comment]
https://github.com/it-muslim/quran-words/pull/24/files/8dc030f2529626db653ae5f6f0dec74382ae8e31#diff-537801bcc83cdd4ef081c66d69147779

File: frontend/angular/src/app/home/home.component.ts (hunk 3 above)
Comment (govza) [reply frontend/angular/src/app/home/home.component.ts:37]: I think I need help with that. Can we talk in the discord group?

**Pull request merged**